### PR TITLE
Separated Database file creation

### PR DESCRIPTION
**Pull request by harshkpatel**
## Motivation and Context

This pull request isolates parts of the code (database file creation) as prior to this change, the database creation happened during the process of parsing course and graph information. Now this allows for more scalability. 

## Your Changes

**Description**: Moved code that already handled database creation to its own function and added migration to that function.

**Type of change** (select all that apply):

- [x] Refactoring (internal change to codebase, without changing functionality)
- [x] Documentation update (change that modifies or updates documentation only)

## Testing

I ran the test suite and got no errors. I also ran through the commands in the README file and everything work accordingly.

## Questions and Comments (if applicable)

One comment I will make is that I get a bus error when running `stack run database`

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CircleCI tests have passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,8 +49,9 @@ $ stack build   # Compile Courseography and all Haskell dependencies (this will
 
 #### Parsing and Generation
 
-1. Create database file for an parse prerequisite graph `$ stack run graphs`
-2. Parse course information `$ stack run database`
+1. Create the database file `$ stack exec courseography database-setup`
```
Comment (david-yz-liu) [inline README.md:52]: You can use `stack run database-setup` here to be consistent with the other commands (both `stack run` and `stack exec courseography` do the same thing for our purposes)

File: README.md [hunk 2]
```diff
@@ -49,8 +49,9 @@ $ stack build   # Compile Courseography and all Haskell dependencies (this will
...
+2. Parse prerequisite graph `$ stack run graphs`
```
Comment (david-yz-liu) [inline README.md:53]: Please change "graph" to "graphs", the original wording was actually misleading!

File: app/Database/Database.hs [hunk 3]
```diff
@@ -25,16 +25,21 @@ breathTableSetUpStr :: String
 breathTableSetUpStr = "breadth table set up"
 
 
--- | Main function for setting up the database with course information.
+-- | Creates the database if it doesn't exist and runs migrations.
+setupDatabase :: IO ()
+setupDatabase = do
+  -- Create db folder if it doesn't exist
```
Comment (david-yz-liu) [inline app/Database/Database.hs:31]: Make sure this comment is indented \to align with the code below

File: app/Main.hs [hunk 4]
```diff
@@ -28,10 +28,11 @@ import DynamicGraphs.WriteRunDot (generatePrereqsForCourses)
 taskMap :: Map.Map String ([String] -> IO ())
 taskMap = Map.fromList [
     ("server", const runServer),
-    ("database", const setupDatabase),
+    ("database", const runDatabase),
```
Comment (david-yz-liu) [inline app/Main.hs:31]: Overall this is a good time to fix up some of the naming of this command and the `graphs` command. Change the following:

- the command `database` to `database-courses`
- the command `graphs` to `database-graphs`
- the function `runDatabase` to `populateCourseInfo`

and make sure to update the README instructions as well!

**Review by david-yz-liu — COMMENTED** [review]
@harshkpatel good work, you've definitely made the right changes. I left a few style/documentation comments and a comment to revise some naming.

**Comment by harshkpatel** [comment]
I have added all the changes you mentioned. I have also run the test suite with no errors and each command in the terminal, and they work accordingly. The workflow test failed, but it was a memory leak issue, which is separate from this task.

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**